packages/client/hmi-client/src/components/widgets/tera-input.vue

```diff
@@ -122,10 +122,8 @@ const unmask = () => {
 	// convert back to a number when finished
 	if (isNistType && !getErrorMessage.value) {
 		emit('update:model-value', nistToNumber(maskedValue.value));
-	}
-
-	if (isTextContainingOnlyDigits(maskedValue.value)) {
-		emit('update:model-value', nistToString(maskedValue.value));
+	} else if (isTextContainingOnlyDigits(maskedValue.value)) {
+		emit('update:model-value', nistToNumber(maskedValue.value));
```

Comment:
Why the reversal, this is needed for expressions?
And if you have to remove it, you should remove the function as well.

Reply:
The component isn't really working as expected at the moment so I was trying some things. Im going to look into this a bit more...this component is not in a great state at the moment

Reply:
Okay, because doing this might create some issue on the model page